homeassistant/components/airnow/types.py

```diff
@@ -0,0 +1,13 @@
+"""The AirNow integration."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from homeassistant.config_entries import ConfigEntry
+
+if TYPE_CHECKING:
+    from .coordinator import AirNowDataUpdateCoordinator
+
+
+type AirNowConfigEntry = ConfigEntry[AirNowDataUpdateCoordinator]
```

Comment:
My recommendations recently have been to move it to `coordinator.py` for this exact reason, but not all approved.

I don't like `if TYPE_CHECKING` imports either - but if it's inside a `types` module I think it's kind of expected.

Reply:
The reason I brought this up is because usually, the runtime data is typed with the coordinator. And then its easier to put it in the coordinator.py. But OTOH, coordinator.py is a file that generally only contains info about how we can fetch data. While the typing of the config entry is something that is more related to runtime. And I kinda view init.py as the setup + runtime management side of things, so it would fit there more. 

And its certainly not a blocker or a must from my side, I was just thinking out loud :)

Reply:
I personally prefer to have it in `__init__.py` as for me this typing is for the integration level.
Also `coordinator.py` doesn't always exist so therefore that's not really an option.
And lastly creating a new file just for a "simple" type seems excessive to me.

Anyhow that didn't help too much either but I guess getting some consistency is good.

Reply:
> I'm uncertain if this is an improvement. 

Agree, which is maybe the best reason not to do it.

>  it often times adds a lot of boilerplate code

For that reason, really. Introducing another module for just this, is overkill.

> I don't like if TYPE_CHECKING imports 

🤷 I think this use case is perfectly valid. 

See also PEP-0563: https://peps.python.org/pep-0563/#runtime-annotation-resolution-and-type-checking

If you are using the full config on ruff it will actually check for that (`TCH004`): 

https://docs.astral.sh/ruff/rules/runtime-import-in-type-checking-block/

I think I would even not be mad if one would put it in `const.py`.

In the end, I'm wonder if we even should add rules for something like this at all. It will raise a bar, that is not really needed.

../Frenck